blame/blame_test.go

```diff
@@ -0,0 +1,810 @@
+		repeat("0d9c9cef53af38cefcb6801bb492aaed3f2c9a42", 1),
+		repeat("a24001f6938d425d0e7504bdf5d27fc866a85c3d", 3),
+		repeat("f66196ceed7d6aeca313b0632657ab762487ced3", 8),
+		repeat("6eb5d9c5225224bfe59c401182a2939d6c27fc00", 1),
+		repeat("f66196ceed7d6aeca313b0632657ab762487ced3", 1),
+		repeat("a24001f6938d425d0e7504bdf5d27fc866a85c3d", 3),
+	)},
+	{"https://github.com/spinnaker/spinnaker.git", "master", "f39d86f59a0781f130e8de6b2115329c1fbe9545", "dev/refresh_source.sh", concat(&[]string{},
+		repeat("a24001f6938d425d0e7504bdf5d27fc866a85c3d", 17),
+	)},
+	/*
+		// FAIL move?
+		{"https://github.com/spinnaker/spinnaker.git", "master", "f39d86f59a0781f130e8de6b2115329c1fbe9545", "dev/create_google_dev_vm.sh", concat(&[]string{},
+			repeat("a24001f6938d425d0e7504bdf5d27fc866a85c3d", 20),
+		)},
+	*/
+}
+
+// run a blame on all the suite's tests
+func (s *SuiteCommon) TestBlame(c *C) {
+	for _, t := range blameTests {
+		fmt.Println("Blamming", t.repo, t.branch, t.commit, t.path)
+		repo, ok := s.repos[t.repo]
+		c.Assert(ok, Equals, true)
+
+		commit, err := repo.Commit(core.NewHash(t.commit))
+		c.Assert(err, IsNil)
+
+		blames, err := Blame(repo, commit, t.path)
+		c.Assert(err, IsNil)
+
+		c.Assert(len(blames), Equals, len(t.blames), Commentf("\nrepo=%s, branch=%s, commit=%s, path=%s",
+			t.repo, t.branch, t.commit, t.path))
+		c.Assert(blames, DeepEquals, s.commits(c, t.repo, t.blames...), Commentf("\nrepo=%s, branch=%s, commit=%s, path=%s, \n%s",
+			t.repo, t.branch, t.commit, t.path, compareSideBySide2(t.blames, blames)))
+	}
+}
+
+// TODO: duplicated from revlist/revlist_test.go
+// returns the commits from a slice of hashes
+func (s *SuiteCommon) commits(cc *C, repo string, hs ...string) []*git.Commit {
+	r, ok := s.repos[repo]
+	cc.Assert(ok, Equals, true)
+	result := make([]*git.Commit, 0, len(hs))
+	for _, h := range hs {
+		c, err := r.Commit(core.NewHash(h))
+		cc.Assert(err, IsNil)
+		result = append(result, c)
+	}
+	return result
+}
+
+// TODO: duplicated from revlist/revlist_test.go
+// same length is assumed
+func compareSideBySide2(a []string, b []*git.Commit) string {
+	var buf bytes.Buffer
+	buf.WriteString("\t              EXPECTED                                          OBTAINED        ")
+	var sep string
+	var obtained string
+	for i := range a {
+		obtained = b[i].Hash.String()
+		if a[i] != obtained {
+			sep = "------"
+		} else {
+			sep = "      "
+		}
+		buf.WriteString(fmt.Sprintf("\n%d", i+1))
+		buf.WriteString(sep)
+		buf.WriteString(a[i])
+		buf.WriteString(sep)
+		buf.WriteString(obtained)
+	}
+	return buf.String()
+}
+
+// utility function to avoid writing so many repeated commits
+func repeat(s string, n int) []string {
+	if n < 0 {
+		panic("repeat: n < 0")
+	}
+	r := make([]string, 0, n)
+	for i := 0; i < n; i++ {
+		r = append(r, s)
+	}
+	return r
+}
+
+// utility function to concat slices
+func concat(dst *[]string, vargs ...[]string) []string {
+	for _, ss := range vargs {
```

Comment:
Apparently all tests use a `&[]string{}` as first input, wouldn't it be easier to just declare a `var dst []string` in here and return that?